Rover/AP_Arming.cpp

```diff
@@ -82,6 +82,11 @@ bool AP_Arming_Rover::pre_arm_checks(bool report)
         return false;
     }
 
+    if (rover.g2.sailboat.sail_enabled() & !rover.g2.windvane.enabled()) {
```

Comment:
&& ?
maybe create a boat_checks could be good

Reply:
fixed, also in the existing function ;) , Sailboats don't actuality have to be boats (despite the name). Happy to put this in a different function, seems abit much for just this one new check tho.

Reply:
These were supposed to be `&`, not `&&` - that way we run all of the checks rather than failing at the first one!

Reply:
@IamPete1, how'd you like to make a small PR to revert the 2nd change?  I think we should add a comment as well explaining why it's single ampersand.